From 385147eaf34b20d0ceed54cc2bf1bd43e595fc75 Mon Sep 17 00:00:00 2001
From: Osamu Aoki <osamu@debian.org>
Date: Sun, 6 Sep 2015 11:28:43 +0900
Subject: [PATCH] Work around with missing debian.tar.xz etc.

This makes our life easy to use uupdate from VCS checkout directory.
---
 scripts/uupdate.sh | 18 ++++++++++++++++++
 1 file changed, 18 insertions(+)

diff --git a/scripts/uupdate.sh b/scripts/uupdate.sh
index e037327..c47a6fe 100755
--- a/scripts/uupdate.sh
+++ b/scripts/uupdate.sh
@@ -723,6 +723,24 @@ else
 	DIFF="../${PACKAGE}_$SVERSION.debian.tar.xz"
 	DIFFTYPE=tar
 	DIFFUNPACK="tar --xz -xf"
+    else
+	# non-native package with $SVERSION in <version>-<revision> format
+	# missing diff.gz/debian.tar.xz, make it with $OPWD/debian/*
+	# force to update next version to 3.0 (quilt)
+	cd $OPWD
+	if [ ! -d debian ]; then
+	    echo "$PROGNAME: None of *.diff.gz, *.debian.tar.xz, or debian/* found. failed;" >&2
+	    echo "aborting..." >&2
+	    exit 1
+	fi
+	mkdir -p debian/source
+	echo "3.0 (quilt)" > debian/source/format
+	tar --xz -cf ../${PACKAGE}_$SVERSION.debian.tar.xz debian
+	# return back to upstream source
+	cd ../$PACKAGE-$SNEW_VERSION
+	DIFF="../${PACKAGE}_$SVERSION.debian.tar.xz"
+	DIFFTYPE=tar
+	DIFFUNPACK="tar --xz -xf"
     fi
 
     if [ "$DIFFTYPE" = diff ]; then
-- 
2.1.4

